crates/client/rpc/src/lib.rs

```diff
@@ -220,9 +226,101 @@ where
     }
 }
 
+#[async_trait]
+impl<A, B, BE, G, C, P, H> StarknetRpcApiCommonFuncs for Starknet<A, B, BE, G, C, P, H>
+where
+    A: ChainApi<Block = B> + 'static,
+    B: BlockT,
+    P: TransactionPool<Block = B> + 'static,
+    BE: Backend<B> + 'static,
+    C: HeaderBackend<B> + BlockBackend<B> + StorageProvider<B, BE> + 'static,
+    C: ProvideRuntimeApi<B>,
+    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
+    G: GenesisProvider + Send + Sync + 'static,
+    H: HasherT + Send + Sync + 'static,
+{
+    async fn declare_txn_common(
+        &self,
+        transaction_inputs: DeclareTransactionCommonInput,
```

Comment:
we can move `DeclareTransactionCommonInput` in `rpc/types` if we remove the trait as mentioned aboved